core/cppgc.rs

```diff
@@ -1,74 +1,88 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 
+use crate::JsRuntime;
 use std::any::TypeId;
+pub use v8::cppgc::GarbageCollected;
 
-pub trait GcResource {}
+const CPPGC_TAG: u16 = 1;
 
-struct CppGcObject<T: GcResource> {
+#[repr(C)]
+struct CppGcObject<T: GarbageCollected> {
   tag: TypeId,
-  member: T,
+  member: v8::cppgc::Member<T>,
 }
 
-impl<T: GcResource> v8::cppgc::GarbageCollected for CppGcObject<T> {
-  fn trace(&self, _: &v8::cppgc::Visitor) {}
+impl<T: GarbageCollected> v8::cppgc::GarbageCollected for CppGcObject<T> {
+  fn trace(&self, visitor: &v8::cppgc::Visitor) {
+    visitor.trace(&self.member);
+  }
 }
 
-pub(crate) const DEFAULT_CPP_GC_EMBEDDER_ID: u16 = 0xde90;
-
-const EMBEDDER_ID_OFFSET: i32 = 0;
-const SLOT_OFFSET: i32 = 1;
-const FIELD_COUNT: usize = 2;
+pub fn make_cppgc_template<'s>(
+  scope: &mut v8::HandleScope<'s, ()>,
+) -> v8::Local<'s, v8::FunctionTemplate> {
+  fn empty(
+    _scope: &mut v8::HandleScope,
+    _args: v8::FunctionCallbackArguments,
+    _rv: v8::ReturnValue,
+  ) {
+  }
+  v8::FunctionTemplate::new(scope, empty)
```

Comment:
Is it possible that a cppgc object will ever end up in the snapshot? If so, this will cause external reference problems, but maybe it's a problem for another day

Reply:
good call, I can add that